### ISSUE TITLE
hardwareBackPress does not seem to be called

### ISSUE THREAD

**Issue by mparpaillon** [facebook/react-native#2887]
Hi,

New react-native dev here.

I'm trying to use the BackAndroid event listener in the exact same way as the MovieApp example and I can't get it to work.

The "hardwareBackPress" event does not seem to be called when I press the back button of my Nexus 5.

Is there any extra configuration that I'm missing ?

Thanks a lot

**Comment by ChristianHersevoort** [comment]
I think it was a bug in the generator, see https://github.com/facebook/react-native/pull/2801 . Try adding the code from this issue to your main activity class

**Comment by mparpaillon** [comment]
Great ! 

Adding this in the MainActivity.java file fixed it for me 

```java
@Override
public void onBackPressed() {
    if (mReactInstanceManager != null) {
        mReactInstanceManager.onBackPressed();
    } else {
        super.onBackPressed();
    }
}
```

**Comment by skyride99** [comment]
That fix will not work with RN 0.19, any other suggestions?
